include/seastar/http/reply.hh

```diff
@@ -200,22 +201,29 @@ struct reply {
     }
 
     /**
-     * Set the content type mime type.
-     * Used when the mime type is known.
-     * For most cases, use the set_content_type
+     * Set the content type. The content_type string can be one of:
+     * 1. A MIME (RFC 2045) Content-Type - looking like "type/subtype", e.g.,
+     *   "text/html"
+     * 2. If a "/" is missing in the given string, we look it up in a list of
+     *    common file extensions listed in the http::mime_types map. For
+     *    example "html" will be mapped to "text/html".
      */
-    reply& set_mime_type(const sstring& mime) {
-        _headers["Content-Type"] = mime;
+    reply& set_content_type(std::string_view content_type) {
```

Comment:
what about `seastar::http::request::set_content_type`?

Reply:
Note that the patch's title, the commit message, and even the 4 year old issue, were all about HTTP server, not http client....

Reply:
I'll add a second patch to this pull request, with a http::request change...

Reply:
right, but the problem is symetric for both. I guess we can do it in separate PR